packages/core/src/fhir-deduplication/resources/observation-shared.ts

```diff
@@ -0,0 +1,118 @@
+import { CodeableConcept, Observation } from "@medplum/fhirtypes";
+import { LOINC_CODE, LOINC_OID, SNOMED_CODE, SNOMED_OID } from "../../util/constants";
+
+export const observationStatus = [
+  "entered-in-error",
+  "unknown",
+  "registered",
+  "preliminary",
+  "final",
+  "amended",
+  "corrected",
+  "cancelled",
+] as const;
+
+export type ObservationStatus = (typeof observationStatus)[number];
+
+export const statusRanking = {
+  unknown: 0,
+  "entered-in-error": 1,
+  registered: 2,
+  preliminary: 3,
+  cancelled: 4,
+  corrected: 5,
+  amended: 6,
+  final: 7,
+};
+
+export function extractCodes(concept: CodeableConcept | undefined): {
+  loincCode: string | undefined;
+  snomedCode: string | undefined;
+  otherCode: string | undefined;
+} {
+  let loincCode = undefined;
+  let snomedCode = undefined;
+  let otherCode = undefined;
+  if (!concept) return { loincCode, snomedCode, otherCode };
+
+  if (concept && concept.coding) {
+    for (const coding of concept.coding) {
+      const system = coding.system?.toLowerCase();
+      const code = coding.code?.trim().toLowerCase();
+      if (system && code) {
+        if (system.includes(LOINC_CODE) || system.includes(LOINC_OID)) {
+          loincCode = code;
+        } else if (system.includes(SNOMED_CODE) || system.includes(SNOMED_OID)) {
+          snomedCode = code;
+        } else {
+          const display = coding.display?.trim().toLowerCase();
+          const text = concept.text?.trim().toLowerCase();
+          if (
+            system.includes("unknown") &&
+            code.includes(unknownCoding.code) &&
+            (display === unknownCoding.display || text === unknownCode.text)
+          ) {
+            // This identifies and ignores unknown codes
+            continue;
+          } else {
+            otherCode = system + code;
+          }
+        }
+      }
+    }
+  }
+  return { loincCode, snomedCode, otherCode };
+}
+
+export function retrieveCode({
+  loincCode,
+  snomedCode,
+  otherCode,
+}: {
+  loincCode: string | undefined;
+  snomedCode: string | undefined;
+  otherCode: string | undefined;
+}): string | undefined {
+  if (loincCode) return loincCode;
+  if (snomedCode) return snomedCode;
+  if (otherCode) return otherCode;
+  return undefined;
+}
+
+export function extractValueFromObservation(observation: Observation) {
```

Comment:
Define the return type of the function, please.

Reply:
Done. I dont really think that's useful tho in this case tbh